### PR TITLE
alternate try at providing modularity to unique ids.

### PR DESCRIPTION
**Pull request by biscuitWizard**
lets modders or downstreams determine how things are generated by overriding uniqueness_generator. assumes all things returned will be strings.

### CODE REVIEW

File: code/datums/repositories/unique.dm [hunk 1]
```diff
@@ -20,6 +22,9 @@ var/repository/unique/uniqueness_repository = new()
 /datum/uniqueness_generator/proc/Generate()
 	return
 
+/datum/uniqueness_generator/hex/Generate()
+	return "[RANDBYTE]-[RANDBYTE]-[RANDBYTE]-[RANDBYTE]"
```
Comment (PsiOmegaDelta) [inline code/datums/repositories/unique.dm:26]: While the probability is exceedingly small that a duplicate will occur I'll note that there's nothing inherent in the system that prevents them.
See `/datum/uniqueness_generator/id_random/Generate()` as an example

Comment (biscuitWizard) [reply code/datums/repositories/unique.dm:26]: I don't want to change CHINSKY's code more than necessary. Resolving that problem is outside of the scope of this PR, I feel.

Comment (PsiOmegaDelta) [reply code/datums/repositories/unique.dm:26]: Merely making a note

Comment (biscuitWizard) [reply code/datums/repositories/unique.dm:26]: Comment rescinded. In that case I agree completely.

**Review by comma — APPROVED** [review]

**Pull request merged**